src/store/plugins/names.js

```diff
@@ -95,15 +97,23 @@ export default (store) => {
 
         commit('set', names);
       },
-      async fetchAuctions({ rootState: { middleware } }) {
+      async fetchAuctions({
+        rootState: { middleware },
+        rootGetters: { activeNetwork },
+      }, { next, filterBy = 'expiration', filterDirection = 'forward' }) {
         if (!middleware) return [];
-        return (
-          await middleware.getAllAuctions({ by: 'expiration', direction: 'forward' })
-        ).data.map(({ name, info }) => ({
-          name,
-          expiration: info.auctionEnd,
-          lastBid: info.lastBid.tx,
-        }));
+        const response = await (next
+          ? fetchJson(`${activeNetwork.middlewareUrl}${next}`)
```

Comment:
I'm proposing to add a fetch-by-path method to middleware: https://github.com/aeternity/superhero-wallet/blob/f1d03c7d1104cc80b38eac9dc379f4c6f4a2c170/src/lib/wallet.js#L73
to extract `fetchJson`, `middlewareUrl`, `camelcaseKeysDeep `